src/main/java/io/antmedia/streamsource/StreamFetcherManager.java

```diff
@@ -45,21 +48,63 @@
 
 	protected AtomicBoolean isJobRunning = new AtomicBoolean(false);
 
-	public StreamFetcherManager(ISchedulingService schedulingService, IDataStore datastore,IScope scope) {
+	public static class StreamFetcherFactory {
+		public StreamFetcher make(Broadcast stream, IScope scope) throws Exception {
+			return new StreamFetcher(stream, scope);
+		}
+	}
+	
+	private boolean restartStreamAutomatically = true;
+
+	public StreamFetcherFactory streamFetcherFactory;
 
+	/**
+	 * Time period in seconds for restarting stream fetchers
+	 */
+	private int restartStreamFetcherPeriodSeconds;
 
+	public StreamFetcherManager(ISchedulingService schedulingService, IDataStore datastore,IScope scope) {
+		this(schedulingService, datastore, scope, null);
+	}
+	
+	
+	
+	public StreamFetcherManager(ISchedulingService schedulingService, IDataStore datastore,IScope scope, StreamFetcherFactory streamFetcherFactory) {
 		this.schedulingService = schedulingService;
 		this.datastore = datastore;
 		this.scope=scope;
+		if (streamFetcherFactory != null) {
+			this.streamFetcherFactory = streamFetcherFactory;
```

Comment:
to make it more readable, this might be an option:
this.streamFetcherFactory = streamFetcherFactory;
if(this.streamFetcherFactory == null){
this.streamFetcherFactory = new StreamFetcherFactory();}

Reply:
check this out 
https://github.com/ant-media/Ant-Media-Server/commit/e3f6052639f925fca08eb81fc8f2f116bd5cce7f